### PR TITLE
fix(FEC-12916): share Plugin Keyboard Functionality

### PR DESCRIPTION
**Pull request by lianbenjamin**
### Description of the Changes

add functions to time-format.js to support https://github.com/kaltura/playkit-js-share/pull/19

Solves FEC-12916

### CheckLists

- [ ] changes have been done against master branch, and PR does not conflict
- [ ] new unit / functional tests have been added (whenever applicable)
- [ ] test are passing in local environment
- [ ] Travis tests are passing (or test results are not worse than on master branch :))
- [ ] Docs have been updated

### CODE REVIEW

File: src/utils/time-format.js [hunk 1]
```diff
@@ -33,15 +33,86 @@ function toHHMMSS(input: number): string {
...
+  return convertSecondsToTimeFormat(val);
+}
+
+/**
+ * Checking whether the value parameter contains only numbers or not
+ *
+ * @param {any} value - the value to check
+ * @returns {boolean} whether the value parameter contains only numbers or not
+ */
+function hasOnlyNumbers(value: any): boolean {
```
Comment (SivanA-Kaltura) [inline src/utils/time-format.js:111]: is !isNan not specific enough ?

File: src/utils/time-format.js [hunk 2]
```diff
@@ -33,15 +33,86 @@ function toHHMMSS(input: number): string {
...
+  return res;
+}
+
+/**
+ * Formatting input that has only numbers to valid time (hh:mm:ss), i.e. 1 to 00:01
+ *
+ * @param {string} val - the value to format
+ * @returns {string} the formatted value
+ */
+function formatOnlyNumbersInput(val: string): string {
```
Comment (SivanA-Kaltura) [inline src/utils/time-format.js:94]: if it's know that it's only numbers at this point, you can convert to numbers and check
val.length < 10
and
val.length < 59

i think that would be clearer

File: src/utils/time-format.js [hunk 3]
```diff
@@ -33,15 +33,86 @@ function toHHMMSS(input: number): string {
  */
 function toSecondsFromHHMMSS(input: string): number {
```
Comment (SivanA-Kaltura) [inline src/utils/time-format.js:34]: if you have separate cases of HHMM and HHMMSS, i think it's better to split into 2 separate functions, 
like toSecondsFromHHMM and toSecondsFromHHMMSS

File: src/utils/time-format.js [hunk 4]
```diff
@@ -33,15 +33,86 @@ function toHHMMSS(input: number): string {
...
+/**
+ * Handling the time parts array
+ *
+ * @param {Array<string>} parts the array
+ * @param {boolean} hasHours whether the parts array contains hours or not
+ * @returns {number} number of seconds
+ * @private
+ */
+function _toSecondsFromArray(parts: Array<string>, hasHours: boolean): number {
+  const minutesIndex = hasHours ? 1 : 0;
```
Comment (SivanA-Kaltura) [inline src/utils/time-format.js:55]: i think that you can shorten it to something like

function _toSecondsFromArray(parts: Array<string>): number {
  let seconds = 0;
  let multiplier = 1;

  for (let i = parts.length - 1; i >= 0; --i) {
    seconds += parts[i] * multiplier;
    multiplier = multiplier * 60;
  }

  return seconds;
}

File: src/utils/time-format.js [hunk 5]
```diff
@@ -33,15 +33,86 @@ function toHHMMSS(input: number): string {
...
 
-export {toHHMMSS, toSecondsFromHHMMSS};
+/**
+ * Converting an input that represents seconds to valid time format (hh:mm:ss), i.e. 62 to 01:02
+ *
+ * @param {string} valInSeconds - the value in seconds to format
+ * @returns {string} the formatted value
+ */
+function convertSecondsToTimeFormat(valInSeconds: string): string {
+  const res = new Date(parseInt(valInSeconds) * 1000).toISOString().substr(11, 8);
```
Comment (SivanA-Kaltura) [inline src/utils/time-format.js:80]: you can call toHHMMSS and cut out the 0's in the result, no ?

**Review by SivanA-Kaltura — CHANGES_REQUESTED** [review]

File: src/utils/time-format.js (hunk 4 above)
Comment (lianbenjamin) [reply src/utils/time-format.js]: cool thanks

**Review by semarche-kaltura — APPROVED** [review]

**Review by SivanA-Kaltura — APPROVED** [review]

**Pull request merged**